hack/log.sh

```diff
@@ -0,0 +1,215 @@
+#!/bin/bash
+#
+# This library holds all of the logging functions for OpenShift bash scripts.
+
+# os::log::install_cleanup installs os::log::clean_up_logger as a trap on interrupts, termination, and exit
+#
+# Globals:
+#  None
+# Arguments:
+#  None
+# Returns:
+#  None
+function os::log::install_cleanup() {
+    trap "os::log::clean_up_logger; $(trap -p SIGINT)" SIGINT
```

Comment:
upstream only does append for traps. if we want this we need to be much more diligent about how we exit traps.

I'll address this in another pull.